### ISSUE TITLE
Execute a Task only if the related predecessors finished executing successfully

### ISSUE THREAD

**Issue by ardialikaj** [cubewise-code/rushti#56]
Can you please add a feature so a certain task, will start only if all related predecessors finished executing successfully

Let's say the below is my example:
id="11" predecessors="" instance="XXfab" process="}bedrock.cube.data.export"
id="21" predecessors="" instance="XXfab" process="}bedrock.cube.data.clear"
id="31" predecessors="11,21" instance="XXfab" process="}bedrock.cube.data.import"

In the example above, task with id="31" will start only after both predecessors are completed. 
There are cases when we do not want to execute a task if any of predecessors failed.
So can you consider implementing a feature so it will be possible to implement a mechanism that the process }bedrock.cube.data.import will be executed only if the predesessors completed successfully

Thank you

**Comment by avanno** [comment]
Hi @ardialikaj,

Quick question before we promote an enhancement for this. Would you like to control each task individually with this new feature? No matter the level of your task?

To illustrate this, let's say you have a new parameter called **'predecessors_succeed'**. This parameter will trigger the execution of the task if their predecessors successfully completed. Now if I follow your example and add a new task with the same level of predecessors that your task "31" as below:

id="11" predecessors="" **predecessors_succeed=""** instance="XXfab" process="}bedrock.cube.data.export"
id="21" predecessors="" **predecessors_succeed=""** instance="XXfab" process="}bedrock.cube.data.clear"
id="31" predecessors="11,21" **predecessors_succeed="1"** instance="XXfab" process="}bedrock.cube.data.import"
id="41" predecessors="11,21" **predecessors_succeed="0"** instance="XXfab" process="}bedrock.cube.data.export"

Both task #31 and #41 belong to the same level of execution: Unlike task "31" which will stop if "11" or "21" failed, task "41" will continue no matter the completion status of their predecessors. Are you ok with this behaviour?

Thank you,

**Comment by ardialikaj** [comment]
> Hi @ardialikaj,
> 
> Quick question before we promote an enhancement for this. Would you like to control each task individually with this new feature? No matter the level of your task?
> 
> To illustrate this, let's say you have a new parameter called **'predecessors_succeed'**. This parameter will trigger the execution of the task if their predecessors successfully completed. Now if I follow your example and add a new task with the same level of predecessors that your task "31" as below:
> 
> id="11" predecessors="" **predecessors_succeed=""** instance="XXfab" process="}bedrock.cube.data.export" id="21" predecessors="" **predecessors_succeed=""** instance="XXfab" process="}bedrock.cube.data.clear" id="31" predecessors="11,21" **predecessors_succeed="1"** instance="XXfab" process="}bedrock.cube.data.import" id="41" predecessors="11,21" **predecessors_succeed="0"** instance="XXfab" process="}bedrock.cube.data.export"
> 
> Both task "31" and "41" belong to the same level of execution: Unlike task "31" which will stop if "11" or "21" failed, task "41" will continue no matter the completion status of their predecessors. Are you ok with this behaviour?
> 
> Thank you,

This is exactly what I was thinking is required.
Thank you

**Comment by MariusWirtz** [comment]
Completed with #57 and #58 

@ardialikaj please confirm that it works as expected.
